pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -57,19 +59,25 @@ type Controller struct {
 	// jLister can list/get Jobs from the shared informer's store
 	jLister batchlisters.JobLister
 
+	// jLister can list/get Pods from the shared informer's store
+	pLister corelisters.PodLister
+
 	// jStoreSynced returns true if the Job store has been synced at least once.
 	// Added as a member to the struct to allow injection for testing.
 	jListerSynced cache.InformerSynced
+	// pStoreSynced returns true if the Pod store has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	pListerSynced cache.InformerSynced
 
-	// Jobs that the controller will check its TTL and attempt to delete when the TTL expires.
+	// Jobs/Pods that the controller will check its TTL and attempt to delete when the TTL expires.
 	queue workqueue.RateLimitingInterface
```

Comment:
you either need a separate queue for jobs and pods, or you need to adjust the key added to the queue to include the object type